src/EA.Weee.Web.Tests.Unit/Areas/Admin/Controllers/CopyAatfControllerTests.cs

```diff
@@ -0,0 +1,311 @@
+        }
+
+        [Fact]
+        public async Task CopyAatfGet_CreatesViewModel_ListsArePopulated()
+        {
+            var aatfData = fixture.Create<AatfData>();
+            aatfData.CanEdit = true;
+            CopyAatfViewModel viewModel = CreateCopyAatfViewModel();
+
+            A.CallTo(() => weeeClient.SendAsync(A<string>._, A<GetAatfById>._)).Returns(aatfData);
+            A.CallTo(() => mapper.Map<CopyAatfViewModel>(aatfData)).Returns(viewModel);
+
+            ViewResult result = await controller.CopyAatfDetails(viewModel.Id) as ViewResult;
+
+            CopyAatfViewModel resultViewModel = result.Model as CopyAatfViewModel;
+
+            Assert.Equal(viewModel.SizeList, resultViewModel.SizeList);
+            Assert.Equal(viewModel.StatusList, resultViewModel.StatusList);
+            Assert.Equal(viewModel.ContactData.AddressData.Countries, resultViewModel.ContactData.AddressData.Countries);
+            Assert.Equal(viewModel.SiteAddressData.Countries, resultViewModel.SiteAddressData.Countries);
+            Assert.Equal(viewModel.OrganisationId, resultViewModel.OrganisationId);
+        }
+
+        [Fact]
+        public async Task CopyAeGet_CreatesViewModel_ListsArePopulated()
+        {
+            var aatfData = fixture.Create<AatfData>();
+            aatfData.FacilityType = FacilityType.Ae;
+            aatfData.CanEdit = true;
+            CopyAeViewModel viewModel = CreateCopyAeViewModel();
+
+            A.CallTo(() => weeeClient.SendAsync(A<string>._, A<GetAatfById>._)).Returns(aatfData);
+            A.CallTo(() => mapper.Map<CopyAeViewModel>(aatfData)).Returns(viewModel);
+
+            ViewResult result = await controller.CopyAatfDetails(viewModel.Id) as ViewResult;
+
+            CopyAeViewModel resultViewModel = result.Model as CopyAeViewModel;
+
+            Assert.Equal(viewModel.SizeList, resultViewModel.SizeList);
+            Assert.Equal(viewModel.StatusList, resultViewModel.StatusList);
+            Assert.Equal(viewModel.ContactData.AddressData.Countries, resultViewModel.ContactData.AddressData.Countries);
+            Assert.Equal(viewModel.SiteAddressData.Countries, resultViewModel.SiteAddressData.Countries);
+            Assert.Equal(viewModel.OrganisationId, resultViewModel.OrganisationId);
+        }
+
+        [Fact]
+        public async void CopyAatfDetailsGet_CanNotEdit_ReturnsForbiddenResult()
+        {
+            var id = fixture.Create<Guid>();
+            var aatf = fixture.Build<AatfData>().With(a => a.CanEdit, false).Create();
+
+            var clientCall = A.CallTo(() => weeeClient.SendAsync(A<string>.Ignored, A<GetAatfById>.That.Matches(a => a.AatfId == id)));
+            clientCall.Returns(aatf);
+
+            var result = await controller.CopyAatfDetails(id);
+
+            Assert.IsType<HttpForbiddenResult>(result);
+            clientCall.MustHaveHappenedOnceExactly();
+        }
+
+        [Fact]
```

Comment:
Need POST tests to ensure that the request is called with the correct aatf id and test to also check that if the model state is not valid the request is called

Reply:
Added